File: 3259/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(maxEnergyBoost([]int{4,1,1}, []int{1,1,3}))
}

func maxEnergyBoost(energyDrinkA []int, energyDrinkB []int) int64 {
	n := len(energyDrinkA)
	d := make([][2]int64, n+1)
	for i := 1; i <= n; i++ {
		d[i][0] = d[i-1][0] + int64(energyDrinkA[i-1])
		d[i][1] = d[i-1][1] + int64(energyDrinkB[i-1])
		if i >= 2 {
			d[i][0] = max(d[i][0], d[i-2][1] + int64(energyDrinkA[i-1]))
			d[i][1] = max(d[i][1], d[i-2][0] + int64(energyDrinkB[i-1]))
		}
	}
	return max(d[n][0], d[n][1])
}

File: 3259/main.rs
struct Solution {}

impl Solution {
    pub fn max_energy_boost(energy_drink_a: Vec<i32>, energy_drink_b: Vec<i32>) -> i64 {
        let n = energy_drink_a.len();
        let mut d = vec![vec![0; 2]; n+ 1];
        for i in 1..=n {
            d[i][0] = d[i-1][0] + energy_drink_a[i-1] as i64;
            d[i][1] = d[i-1][1] + energy_drink_b[i-1] as i64;
            if i >= 2 {
                d[i][0] = d[i][0].max(d[i-2][1] + energy_drink_a[i-1] as i64);
                d[i][1] = d[i][1].max(d[i-2][0] + energy_drink_b[i-1] as i64);
            }
        }
        d[n][0].max(d[n][1])
    }
}

fn main() {
    println!("{}", Solution::max_energy_boost(vec![4,1,1], vec![1,1,3]));
}